hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsTopic.kt

```diff
@@ -11,97 +12,101 @@ import software.amazon.awssdk.services.sns.model.MessageAttributeValue
+    }.onFailure {
+      log.error("""Unable to publish {} with body "{}"""", eventType, event)
+    }.getOrThrow()
+  }
 
-/**
- * Publishes an event to the specified SNS topic with retry and back off policies.
- *
- * Examples:
- *
- * Scenario where API is called from UI, and we want a reasonable number of retries while the user waits for a response:
- * We can just use the default of 4 attempts that would block the thread for around 7 seconds
- * ```
- * hmppsDomainTopic.publish(
- *         eventType = event.eventType,
- *         event = event.body,
- *       )
- * ```
- *
- * Scenario where retrying is already done by the calling code; for instance an SQS Listener that transforms events that is idempotent so the entire process can be rerun:
- * Given we don't want to tie up the listener thread there is no advantage of retrying on this thread so
- *
- * ```
- *    hmppsDomainTopic.publish(
- *       eventType = event.eventType,
- *       event = event.body,
- *       retryPolicy = NeverRetryPolicy(),
- *     )
- * ```
- *
- *
- * Scenario where thread can hang for longer with more retries
- * ```
- * hmppsDomainTopic.publish(
- *         eventType = event.eventType,
- *         event = event.body,
- *         retryPolicy = SimpleRetryPolicy().apply { maxAttempts = 10 }
- *       )
- * ```
- *
- * Scenario where we want to retry many times but use no more than 10 seconds
- * ```
- * hmppsDomainTopic.publish(
- *         eventType = event.eventType,
- *         event = event.body,
- *         retryPolicy = SimpleRetryPolicy().apply { maxAttempts = 20 },
- *         backoffPolicy = FixedBackOffPolicy().apply { backOffPeriod = 500 },
- *       )
- * ```
- *
- * Scenario where we need to set more SNS attributes
- * ```
- * hmppsDomainTopic.publish(
- *         eventType = event.eventType,
- *         event = event.body,
- *         attributes = mapOf(
- *           "fruit" to MessageAttributeValue.builder().dataType("String").stringValue("banana").build(),
- *           "eventType" to MessageAttributeValue.builder().dataType("String").stringValue(event.eventType).build(),
- *         ),
- *        )
- * ```
- *
- * @param eventType The type of the event, which clients listen to, e.g., prisoner.movement.added.
- * @param event The event data in JSON format as a string.
- * @param attributes A map of additional message attributes. By default, the eventType is added as an attribute, so only supply this if you need additional attributes, in which case you would also need to supply the eventType manually.
- * @param retryPolicy The policy for retrying the publish request in case of failure. By default, this will retry 3 times after an error.
- * @param backOffPolicy The policy for backing off between retries. By default, this will be an exponential policy starting at 1 second with a multiplier of 2
- * @return A response from the publish request containing details such as the message ID.
- */
-fun HmppsTopic.publish(
```

Comment:
this didn't need to be an extension function as the class definition was just above anyway

Reply:
It doesn't need to be, but @andymarke did that on purpose.